src/AlarmClock.h

```diff
@@ -56,87 +72,71 @@ template<typename Duration> class AlarmClock {
 
 protected:
 
-   void AlarmClockThread() {
-      SleepTimeIsBelow500ms() ? AlarmClock::SleepForFullAmount() : AlarmClock::SleepInIntervals();
+   void AlarmClockInterruptableThread() {
+      do {
+         mSleepFunction(kSleepTimeUsCount);
+         // This is required because the unit tests use their own "fake sleep"
+         // pass it into the constructor. By doing that, the unit tests will
+         // fail if the mExpired.store(true) is in the sleep function instead
+         // of the thread function. aka Due to unit test limitation.
+         if (!mExit && !mReset) {
+            mExpired.store(true);
+         }
+         if (mExit) {  
+            break;
+         }
+
+         if (mExpired) {
+            while (!mReset && !mExit) {
+               this_thread::sleep_for(chrono::microseconds(1));
+            }
+         }
+         mReset.store(false);
+      } while (!mExit);
    }
   
    void StopBackgroundThread() {
```

Comment:
Can this function be deleted and the join code added directly to the destructor. The object is not really useful if the thread has been joined.